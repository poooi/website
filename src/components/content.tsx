import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components/macro'

import { ErrorBoundary } from './error-boundary'
import { TypeCat } from './type-cat'

import poi from '../assets/poi.png'

import Download from './download'
import IconLoader from './icon-loader'

const Container = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
`

const Title = styled.h1`
  display: flex;
  align-items: center;
  justify-content: center;
  font-weight: normal;
`

const Name = styled.span`
  font-size: 100px;

  @media screen and (min-width: 768px) {
    font-size: 200px;
  }
`

const Description = styled.div`
  font-size: 36px;
  text-align: center;
  white-space: pre-line;

  @media screen and (min-width: 768px) {
    font-size: 48px;
  }
`

const Image = styled.img`
  transition: 0.3s;
  position: absolute;
`

const ImageContainer = styled.div`
  position: relative;

  height: 150px;
  width: 150px;

  img {
    height: 150px;
    width: 150px;
  }

  @media screen and (min-width: 768px) {
    width: 300px;
    height: 300px;

    img {
      height: 300px;
      width: 300px;
    }
  }
`

const Logo = () => {
  return (
    <ImageContainer>
      <Image src={poi} />
    </ImageContainer>
  )
}

export const Content = () => {
  const { t } = useTranslation()
  return (
    <Container>
      <ErrorBoundary>
        <Title>
          <Logo />
          <Name>{t('name')}</Name>
        </Title>
        <Description>
          <TypeCat text={t('description')} />
        </Description>
        <IconLoader />
        <Download />
      </ErrorBoundary>
    </Container>
  )
}
