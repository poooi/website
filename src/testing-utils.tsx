/* eslint-disable import/no-extraneous-dependencies */

import { render, RenderResult } from '@testing-library/react'
import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components/macro'
import { lightTheme } from './theme'

export const renderWithTheme = (node: ReactElement): RenderResult => {
  const result = render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
  )
  return {
    ...result,
    rerender: (newNode: ReactElement) =>
      result.rerender(
        <ThemeProvider theme={lightTheme}>{newNode}</ThemeProvider>,
      ),
  }
}
