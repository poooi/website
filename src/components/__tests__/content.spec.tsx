import { fireEvent } from '@testing-library/react'
import { waitFor } from '@testing-library/dom'
import React from 'react'
import { renderWithTheme } from '../../testing-utils'

import { Content } from '../content'

jest.mock('ua-parser-js')

describe('<Content /> with beta > stable', () => {
  let fetch: jest.SpyInstance
  beforeAll(() => {
    fetch = jest.spyOn(window, 'fetch').mockImplementation(
      async () =>
        new Response(
          JSON.stringify({
            betaVersion: 'v10.5.0-beta.0',
            version: 'v10.4.0',
          }),
        ),
    )
  })

  afterAll(() => {
    fetch.mockReset()
  })

  it('renders', async () => {
    const { asFragment, getByTestId } = renderWithTheme(<Content />)
    await waitFor(() => getByTestId('download-stable-version'))
    expect(asFragment()).toMatchSnapshot()
  })

  it('opens target list', async () => {
    const { getByTestId, baseElement } = renderWithTheme(<Content />)
    await waitFor(() => getByTestId('download-stable-version'))
    fireEvent.click(getByTestId('open-dialog'))
    await waitFor(() => getByTestId('modal-container'))
    expect(
      baseElement.querySelector('[data-testid="modal-container"]'),
    ).toMatchSnapshot()
  })
})

describe('<Content /> with beta < stable', () => {
  let fetch: jest.SpyInstance
  beforeAll(() => {
    fetch = jest.spyOn(window, 'fetch').mockImplementation(
      async () =>
        new Response(
          JSON.stringify({
            betaVersion: 'v10.3.0-beta.0',
            version: 'v10.4.0',
          }),
        ),
    )
  })

  afterAll(() => {
    fetch.mockReset()
  })

  it('renders', async () => {
    const { asFragment, getByTestId } = renderWithTheme(<Content />)
    await waitFor(() => getByTestId('download-stable-version'))
    expect(asFragment()).toMatchSnapshot()
  })

  it('opens target list', async () => {
    const { getByTestId, baseElement } = renderWithTheme(<Content />)
    await waitFor(() => getByTestId('download-stable-version'))
    fireEvent.click(getByTestId('open-dialog'))
    await waitFor(() => getByTestId('modal-container'))
    expect(
      baseElement.querySelector('[data-testid="modal-container"]'),
    ).toMatchSnapshot()
  })
})
