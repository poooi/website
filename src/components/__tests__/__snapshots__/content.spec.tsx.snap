// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> with beta < stable renders 1`] = `
<DocumentFragment>
  .c6 {
  display: inline-block;
  margin: 20px 10px;
}

.c7 {
  opacity: 1;
}

.c7 .Cursor--blinking {
  opacity: 1;
  -webkit-animation: blink 1s linear infinite;
  animation: blink 1s linear infinite;
}

.c8 {
  opacity: 0;
  display: none;
}

.c8 .Cursor--blinking {
  opacity: 1;
  -webkit-animation: blink 1s linear infinite;
  animation: blink 1s linear infinite;
}

.c9 {
  text-align: center;
  min-height: 150px;
}

.c9 .ms-Spinner-label {
  font-size: 20px;
}

.c11 {
  display: block;
  font-size: 1rem;
}

.c10 {
  padding: 1rem 2em;
  font-size: 1.5rem;
}

.c10.c10 {
  background-color: #F5F8FA;
  color: #182026;
  background-image: none;
  border: 1px solid #182026;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1rem auto;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: normal;
}

.c4 {
  font-size: 100px;
}

.c5 {
  font-size: 36px;
  text-align: center;
  white-space: pre-line;
}

.c3 {
  -webkit-transition: 0.3s;
  transition: 0.3s;
  position: absolute;
}

.c2 {
  position: relative;
  height: 150px;
  width: 150px;
}

.c2 img {
  height: 150px;
  width: 150px;
}

@media screen and (min-width:768px) {
  .c4 {
    font-size: 200px;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    font-size: 48px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    width: 300px;
    height: 300px;
  }

  .c2 img {
    height: 300px;
    width: 300px;
  }
}

<div
    class="c0"
  >
    <h1
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          class="c3"
          src="poi.png"
        />
      </div>
      <span
        class="c4"
      >
        poi
      </span>
    </h1>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <div
            data-testid="typist"
          />
        </div>
        <div
          class="c8"
          data-testid="typing"
        >
          Scalable KanColle browser and tool.
          <span
            style="opacity: 0;"
          >
            _
          </span>
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <a
        class="bp3-button bp3-intent-primary c10"
        data-testid="download-stable-version"
        href="https://npm.taobao.org/mirrors/poi/v10.4.0/poi-setup-10.4.0.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp3-button-text"
        >
          <span>
            Download v10.4.0
          </span>
          <span
            class="c11"
          >
            Windows (x64)
          </span>
          <span
            class="c11"
          >
            Stable, for most users
          </span>
        </span>
      </a>
    </div>
    <div
      class="c12"
    >
      <button
        class="bp3-button bp3-minimal"
        data-testid="open-dialog"
        href="/downloads"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Choose another platform
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Content /> with beta > stable renders 1`] = `
<DocumentFragment>
  .c6 {
  display: inline-block;
  margin: 20px 10px;
}

.c7 {
  opacity: 1;
}

.c7 .Cursor--blinking {
  opacity: 1;
  -webkit-animation: blink 1s linear infinite;
  animation: blink 1s linear infinite;
}

.c8 {
  opacity: 0;
  display: none;
}

.c8 .Cursor--blinking {
  opacity: 1;
  -webkit-animation: blink 1s linear infinite;
  animation: blink 1s linear infinite;
}

.c9 {
  text-align: center;
  min-height: 150px;
}

.c9 .ms-Spinner-label {
  font-size: 20px;
}

.c11 {
  display: block;
  font-size: 1rem;
}

.c10 {
  padding: 1rem 2em;
  font-size: 1.5rem;
}

.c10.c10 {
  background-color: #F5F8FA;
  color: #182026;
  background-image: none;
  border: 1px solid #182026;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1rem auto;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: normal;
}

.c4 {
  font-size: 100px;
}

.c5 {
  font-size: 36px;
  text-align: center;
  white-space: pre-line;
}

.c3 {
  -webkit-transition: 0.3s;
  transition: 0.3s;
  position: absolute;
}

.c2 {
  position: relative;
  height: 150px;
  width: 150px;
}

.c2 img {
  height: 150px;
  width: 150px;
}

@media screen and (min-width:768px) {
  .c4 {
    font-size: 200px;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    font-size: 48px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    width: 300px;
    height: 300px;
  }

  .c2 img {
    height: 300px;
    width: 300px;
  }
}

<div
    class="c0"
  >
    <h1
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          class="c3"
          src="poi.png"
        />
      </div>
      <span
        class="c4"
      >
        poi
      </span>
    </h1>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <div
            data-testid="typist"
          />
        </div>
        <div
          class="c8"
          data-testid="typing"
        >
          Scalable KanColle browser and tool.
          <span
            style="opacity: 0;"
          >
            _
          </span>
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <a
        class="bp3-button bp3-intent-primary c10"
        data-testid="download-stable-version"
        href="https://npm.taobao.org/mirrors/poi/v10.4.0/poi-setup-10.4.0.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp3-button-text"
        >
          <span>
            Download v10.4.0
          </span>
          <span
            class="c11"
          >
            Windows (x64)
          </span>
          <span
            class="c11"
          >
            Stable, for most users
          </span>
        </span>
      </a>
      <a
        class="bp3-button c10"
        href="https://npm.taobao.org/mirrors/poi/v10.5.0-beta.0/poi-setup-10.5.0-beta.0.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp3-button-text"
        >
          <div>
            Download v10.5.0-beta.0
          </div>
          <span
            class="c11"
          >
            Windows (x64)
          </span>
          <span
            class="c11"
          >
            Beta, for new features
          </span>
        </span>
      </a>
    </div>
    <div
      class="c12"
    >
      <button
        class="bp3-button bp3-minimal"
        data-testid="open-dialog"
        href="/downloads"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Choose another platform
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
